homeassistant/components/nuki/__init__.py

```diff
@@ -40,12 +48,43 @@
 
 PLATFORMS = [Platform.BINARY_SENSOR, Platform.LOCK, Platform.SENSOR]
 UPDATE_INTERVAL = timedelta(seconds=30)
+CALLBACK_URL = f"/api/{DOMAIN}/callback"
 
 
 def _get_bridge_devices(bridge: NukiBridge) -> tuple[list[NukiLock], list[NukiOpener]]:
     return bridge.locks, bridge.openers
 
 
+def _register_webhook(bridge: NukiBridge, entry_id: str, url: str) -> bool:
+    # Register HA URL as webhook if not already
+    callbacks = bridge.callback_list()
+    for callback in callbacks["callbacks"]:
+        if entry_id in callback["url"]:
+            if callback["url"] == url:
+                return True
+            bridge.callback_remove(callback["id"])
+
+    if bridge.callback_add(url)["success"]:
+        return True
+
+    return False
+
+
+def _remove_webhook(bridge: NukiBridge, entry_id: str) -> bool:
+    # Remove webhook if set
+    callbacks = bridge.callback_list()
+    for callback in callbacks["callbacks"]:
+        if entry_id in callback["url"]:
+            bridge.callback_remove(callback["id"])
+    return True
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the Nuki component."""
+    hass.http.register_view(NukiCallbackView)
```

Comment:
We have a webhook integration, see i.e. ecowitt. Please use the webhook feature together with this callback

Reply:
Changed to the webhook integration like ecowitt.